### DISCUSSION TITLE
Create global image path

### DISCUSSION THREAD

**Discussion by Int-Circuit** [sphinx-doc/sphinx#12747] [Q&A]
Hello everyone, I'm currently in the process of writing the documentation for my app.

it includes a lot of screenshots and I've wondered if I could create a global link to the image folder.

Example:

Ive got the image folder in source/images, I want to set a link with a identifier so I could write *Image_folder_identifier*/app login screenshot.png

Is it possible?

Thanks in advance

**Comment by chrisjsewell** [comment]
Just an FYI, you know that if you start an image path with `/` it is already relative to the source folder, i.e. you can do:

```restructuredtext
.. image:: /images/image.png
```

**Comment by Int-Circuit** [comment]
@chrisjsewell 
I know that, but for my project, the image path is like this:

"source/_external/_images/folio"/folios properties.PNG

The part in apostrophe is the one I want to replace by a global identifier.

Le 7 août 2024 23:22:47 GMT+02:00, Chris Sewell ***@***.***> a écrit :

**Comment by cederom** [comment]
+1 to be able to add image search paths.. this would be very useful for modular documentation from different places, for instance different README.rst that display images on git web repos while browsing, while the sphinx root is located somewhere else :-)

**Comment by gastmaier** [comment]
i recommend making a sphinx transform for that.
this does not sound something that should be a core feature.
I have something 'similar' at 
https://github.com/analogdevicesinc/doctools/blob/main/adi_doctools/transforms/lfs.py

where is replace git lfs pointers with a github raw url, to not upload 'again' and due to gh pages does not support lfs anyway

**Comment by SilverRainZ** [comment]
Recently, I wrote an extension that supports flexible template rendering in Sphinx. It should meet your needs.

```shell
pip install "sphinxnotes-render[ext]"
```

Add the following code to your `conf.py`:

```py
render_ext_data_define_directives = {
   'screenshots': {
       'schema': {
           'name': 'str, required',
       },
       'template': {
           'text': '.. image:: source/_external/_images/folio/{{ name }}.png',
       },
   },
}
```

the above config create a directive called ``screenshot``, When you use ``.. screenshot:: app`` in your document, it will be rendered to
``.. image:: source/_external/_images/folio/app.png`` and then parsed by Sphinx.

For more details, please refer to https://github.com/sphinx-notes/render and https://sphinx.silverrainz.me/render/
